File: ui/lib/api/agent-skill.ts
import { apiClient } from "./client";
import type { ApiSuccess } from "./auth";

export interface AgentSkillInfo {
  id: string;
  name: string;
  label: string;
  description: string | null;
  entrypoint: string;
  keywords: string[];
  config: Record<string, unknown>;
  enabled: boolean;
  sort: number;
  created_at: string;
  updated_at: string | null;
}

export function listAgentSkills() {
  return apiClient.get<ApiSuccess<AgentSkillInfo[]>>("/agent-skills");
}

export function updateAgentSkill(id: string, body: Partial<AgentSkillInfo>) {
  return apiClient.put<ApiSuccess<AgentSkillInfo>>(`/agent-skills/${id}`, body);
}

export function routeAgentSkill(prompt: string) {
  return apiClient.post<ApiSuccess<AgentSkillInfo | null>>("/agent-skills/route", { prompt });
}

File: backend/app/schemas/agent_skill.py
"""Agent skill schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AgentSkillInfo(BaseModel):
    id: UUID
    name: str
    label: str
    description: str | None = None
    entrypoint: str
    keywords: list[str]
    config: dict
    enabled: bool
    sort: int
    created_at: datetime
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class AgentSkillCreate(BaseModel):
    name: str
    label: str
    description: str | None = None
    entrypoint: str
    keywords: list[str] | None = None
    config: dict | None = None
    enabled: bool | None = True
    sort: int | None = 100


class AgentSkillUpdate(BaseModel):
    name: str | None = None
    label: str | None = None
    description: str | None = None
    entrypoint: str | None = None
    keywords: list[str] | None = None
    config: dict | None = None
    enabled: bool | None = None
    sort: int | None = None


class SkillRouteRequest(BaseModel):
    prompt: str
